pkg/kv/store.go

```diff
@@ -41,18 +41,22 @@ type Driver interface {
 	Open(ctx context.Context, dsn string) (Store, error)
 }
 
+type Predicate interface{}
+
 type Store interface {
 	// Get returns a value for the given key, or ErrNotFound if key doesn't exist
 	Get(ctx context.Context, key []byte) ([]byte, error)
 
+	GetEntry(ctx context.Context, key []byte) (*Entry, error)
+
```

Comment:
Renamed to GetValuaPredicate - the case where Get without predicate is a lighter version of Get with predicate.
I've tested the "options" variant, where we have one Get and one Set method and passing options to the function control if you like to pass a predicate or not.
The problem I see in this style is the contract that the implementor need to address - the combination of all the options and the expected result.
When you have a method signature - it is easy to pass the requirement for the implementor.

Reply:
I get why we would like to have 2 `Set`s functions, since the predicate existence changes the behaviour.
I don't get why there's a need for 2 `Get`s. We pass the predicate and if the caller wish, she can ignore it. It's not saving any time/memory, just makes the api more confusing..

Reply:
Didn't want to lock the option that there can be a version of 'get' with predicate that will slower to use than a 'get' for the latest value (without predicate).
I'll discover and see how we plan to implement the 'get' for different underlying storage types - if it is the same cost as calling a simple 'get' - I'll make predicate optional and leave one API.